### PR TITLE
[1578]-Add new role CAS3_REPORTER to access report in all region

### PR DESCRIPTION
**Pull request by sudhakar-plr**
# Changes in this PR
See the ticket for more information: [1578](https://trello.com/c/er6PWu29/1578-cas3reporter-role-across-regions)

- Adding new role 'CAS3_REPORTER' to view/download the report for temporary accommodation user
- User with role should able to view/download the report in all region
- Profile API will return TemporaryAccommodationUserRole as 'reporter'
- We will use existing seeding mechanism to create/update this role to the Users.
- Default user 'JIMSNOWLDAP' will have this role added in test,dev env.

Once this is done, then we need to play card: [1646](https://trello.com/c/1yScwtsf/1646-remove-type-casts-for-reporter-role) to tidy-up front end.

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

**Review by tahb — COMMENTED** [review]
Hey Sudhakar 👋 When I get more time today i'd like to give this a test locally.

Whilst I do that I wanted to ask that we [break our commits up into small atomic pieces](https://playbook.dxw.com/tech/how-we-do-development/#5-commit-changes-in-small-steps). The team standard from the onboarding checklist is worth reading through. The pull request description is super and there's a actually a list of the logical parts to this change that may help us separate out these commits:

-     User with role should able to view/download the report in all region
-     Profile API will return TemporaryAccommodationUserRole as 'reporter'
-     We will use existing seeding mechanism to create/update this role to the Users.
-     Default user 'JIMSNOWLDAP' will have this role added in test,dev env.

**Comment by sudhakar-plr** [comment]
> Hey Sudhakar 👋 When I get more time today i'd like to give this a test locally.
> 
> Whilst I do that I wanted to ask that we [break our commits up into small atomic pieces](https://playbook.dxw.com/tech/how-we-do-development/#5-commit-changes-in-small-steps). The team standard from the onboarding checklist is worth reading through. The pull request description is super and there's a actually a list of the logical parts to this change that may help us separate out these commits:
> 
> * User with role should able to view/download the report in all region
> * Profile API will return TemporaryAccommodationUserRole as 'reporter'
> * We will use existing seeding mechanism to create/update this role to the Users.
> * Default user 'JIMSNOWLDAP' will have this role added in test,dev env.

Reorganised the commit

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/transformer/UserTransformerTest.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      temporaryAccommodation,
+    ) as TemporaryAccommodationUser
+
+    assertThat(result.roles).contains(reporter)
+    assertThat(result.service).isEqualTo(temporaryAccommodation.value)
+    verify(exactly = 1) { probationRegionTransformer.transformJpaToApi(any()) }
+  }
+
+  @Test
+  fun `Should successfully transfer user entity with role CAS3_REFERRER to referrer`() {
```
Comment (tahb) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/transformer/UserTransformerTest.kt:52]: Thank you for retroffitting this test file and writing tests for other roles! 🙌

**Review by tahb — COMMENTED** [review]
Looks good to me 👍 I've finally managed to get my local environment fixed and was able to verify it looks good from the frontend perspective.

@danlivings-dxw can you cast your more experienced Kotlin eye over this too please

**Review by danlivings-dxw — APPROVED** [review]

**Pull request merged**